src/index/PatternIndex.cpp

```diff
@@ -0,0 +1,834 @@
+#include "PatternIndex.h"
+#include "MetaDataIterator.h"
+
+const uint32_t PatternIndex::PATTERNS_FILE_VERSION = 1;
+const uint32_t PatternIndex::NAMESPACE_FILE_VERSION = 0;
+
+const std::string PatternIndex::PATTERNS_NAMESPACE_SUFFIX =
+    ".index.patterns.pred_namespace";
+const std::string PatternIndex::PATTERNS_SUBJECTS_SUFFIX = ".index.patterns";
+const std::string PatternIndex::PATTERNS_OBJECTS_SUFFIX =
+    ".index.patterns.objects";
+
+// TODO: stxxl::vector<array<Id, 3>> should be Index::TripleVec but that
+// creates a circular dependency
+using TripleVec = stxxl::vector<array<Id, 3>>;
```

Comment:
I think there is a File `IndexBuilderTypes.h` or similar for this reasons. But in this case I am fine with it.